index.js

```diff
@@ -71,27 +75,41 @@ if(!fs.existsSync(path.join(process.env.DATABASE, 'cache'))) {
 if(!fs.existsSync(path.join(process.env.DATABASE, 'cache','images'))) {
     fs.mkdirSync(path.join(process.env.DATABASE, 'cache','images'))
 }
+if(!fs.existsSync(pluginsFolder)) {
+    fs.mkdirSync(pluginsFolder)
+}
 
 
 channelDB = new ChannelDB( path.join(process.env.DATABASE, 'channels') );
 fillerDB = new FillerDB( path.join(process.env.DATABASE, 'filler') , channelDB, channelCache );
 
 customShowDB = new CustomShowDB( path.join(process.env.DATABASE, 'custom-shows') );
 
-db.connect(process.env.DATABASE, ['channels', 'plex-servers', 'ffmpeg-settings', 'plex-settings', 'xmltv-settings', 'hdhr-settings', 'db-version', 'client-id', 'cache-images', 'settings'])
+db.connect(process.env.DATABASE, ['channels', 'plex-servers', 'ffmpeg-settings', 'plex-settings', 'xmltv-settings', 'hdhr-settings', 'db-version', 'client-id', 'cache-images', 'settings', 'plugin-manager'])
```

Comment:
I have multiple concerns here. The method to rely on storage to detect reboots, seems odd. Then we have the usage of the settings db, which really shouldn't be used for that. Finally, I am not sure why do we need this hook? Since dizquetv is always running, then you can use a server start hook to cover all of these use cases (when the server starts, you can basically assume it has been restarted)

If you really do have ideas that require a restart trigger specifically that wouldn't work with a start trigger, then tell me.

Reply:
Let me contextualize you.

We need to dizquetv be restarted after an installation or uninstallation of plugins.

Because plugin system use node `require()` to import all plugins, after a plugin is removed, they stay on node memory and broken the server.

My initial Idea, are to make the server restart, but, to make this, I need to create a script to start before the server (express), and store or manager server process.

In the future, we need to make this, for a good practice and redundancy, but for now, to make this is required a big refac on index.js

Look to fix this problem, in a fast way, I created a boolean setting `serverRestart`, to be true when a plugin is installed/unstalled.
With this variable, I respond on router getNeedRestart if serve need to be restarted or not.

This router is called from frontend. When the value is true, the frontend show an alert with a message to user to make a restart on server.
![image](https://user-images.githubusercontent.com/4449232/126878959-d3bf44f0-85c8-4734-83d9-c3f1b2387650.png)

I can't find a way to store this variable to be changed and used from 3 different scripts, without create a singleton to only this. And I don't know if create a singleton, just for this, just for 1 variable, is a good practice.

I thought it might be more reliable to store this information in the DB.

If you believe this is a bad way, You can help me find another way? 😄